x-pack/legacy/plugins/ml/public/application/jobs/components/custom_url_editor/utils.d.ts

```diff
@@ -4,7 +4,40 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { KibanaUrlConfig } from '../../../../../common/types/custom_urls';
+import { IIndexPattern } from 'src/plugins/data/common';
+import { UrlConfig } from '../../../../../common/types/custom_urls';
 import { Job } from '../../new_job/common/job_creator/configs';
+import { TimeRangeType } from './constants';
 
-export function getTestUrl(job: Job, customUrl: KibanaUrlConfig): Promise<string>;
+export interface TimeRange {
+  type: TimeRangeType;
+  interval: string;
+}
+
+export interface CustomUrlSettings {
+  label: string;
+  type: string;
+  // Note timeRange is only editable in new URLs for Dashboard and Discover URLs,
+  // as for other URLs we have no way of knowing how the field will be used in the URL.
+  timeRange: TimeRange;
+  kibanaSettings?: any;
+  otherUrlSettings?: {
+    urlValue: string;
+  };
+}
+
+export function getTestUrl(job: Job, customUrl: UrlConfig): Promise<string>;
```

Comment:
There is a bug in the `getTestUrl` function that this has exposed. For the check for datafeeds that don't use aggregations, `if (response.hits.total > 0)` needs to be `if (response.hits.total.value > 0)`.

Reply:
Fixed in https://github.com/elastic/kibana/pull/51281/commits/dd8f58c53d3fa5d48d9c3adf09870fba48481c46